modes/speedbar/evil-collection-speedbar.el

```diff
@@ -0,0 +1,105 @@
+;;; evil-collection-speedbar.el --- Evil bindings for speedbar -*- lexical-binding: t -*-
+
+;; Copyright (C) 2020 Zhiwei Chen
+
+;; Author: Zhiwei Chen <condy091@gmail.com>
+;; Maintainer: James Nguyen <james@jojojames.com>
+;; Pierre Neidhardt <mail@ambrevar.xyz>
+;; URL: https://github.com/emacs-evil/evil-collection
+;; Version: 0.0.1
+;; Package-Requires: ((emacs "25.1"))
+;; Keywords: evil, speedbar, tools
+
+;; This program is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+;;; Commentary:
+;; Evil bindings for speedbar.
+
+;;; Code:
+
+(require 'evil-collection)
+(require 'speedbar)
+
+(defconst evil-collection-speedbar-maps '(speedbar-mode-map
+                                          speedbar-file-key-map
+                                          speedbar-buffers-key-map))
+
+;;;###autoload
+(defun evil-collection-speedbar-setup ()
+  "Set up `evil' bindings for `speedbar'."
+  (evil-set-initial-state 'speedbar-mode 'normal)
+  (evil-collection-define-key 'normal 'speedbar-mode-map
+    "q" 'dframe-close-frame
+    "j" 'speedbar-next
+    "k" 'speedbar-prev
+    [mouse-2] 'dframe-click
```

Comment:
Same with this one, (I don't use speedbar)

Reply:
The full `speedbar-mode-map`

```
key             binding
---             -------

C-x             Prefix Command
ESC             Prefix Command
Q               delete-frame
b               ??
f               ??
g               speedbar-refresh
n               speedbar-next
p               speedbar-prev
q               dframe-close-frame
r               ??
t               speedbar-toggle-updates
<C-down-mouse-1>                dframe-hack-buffer-menu
<S-mouse-2>                     dframe-power-click
<down-mouse-3>                  dframe-popup-kludge
<follow-link>                   mouse-face
<mode-line>                     Prefix Command
<mouse-2>       dframe-click
<mouse-movement>                dframe-track-mouse
<override-state>                all
<remap>         Prefix Command

C-x b           dframe-switch-buffer-attached-frame

C-M-n           speedbar-forward-list
C-M-p           speedbar-backward-list
M-n             speedbar-restricted-next
M-p             speedbar-restricted-prev

<mode-line> <down-mouse-1>      dframe-popup-kludge
<mode-line> <mouse-2>           dframe-mouse-hscroll
```